src/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -76,7 +76,11 @@ public static int CurrentManagedThreadId
         {
             get
             {
+#if PLATFORM_UNIX
                 return ManagedThreadId.Current;
+#else
+                return unchecked((int)Interop.mincore.GetCurrentThreadId());
```

Comment:
Since GetCurrentThreadId is on perf critical path for Windows for RyuJIT, we should switch `mincore.GetCurrentThreadId()` from DllImport to RuntimeImport to avoid the PInvoke transitions for it. Similar to what was done for GetLastError/SetLastError here: https://github.com/dotnet/corert/pull/2452/files#diff-783dd9c222e5eab3c7d3230dcf6a4c83L11